core/src/main/java/org/elasticsearch/index/query/AndQueryBuilder.java

```diff
@@ -78,7 +100,66 @@ protected void doXContent(XContentBuilder builder, Params params) throws IOExcep
     }
 
     @Override
+    public Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException {
+        if (filters.isEmpty()) {
+            // no filters provided, this should be ignored upstream
+            return null;
```

Comment:
same as comments in the previous PR, should we be ever returning null here? I tend to think we shouldn't.